### PR TITLE
[Auth] #378 auth client

### PR DESCRIPTION
**Pull request by llbartekll**
- Implements Auth Client's methods
- add cleanup service for debugging

### CODE REVIEW

File: Sources/Auth/AuthClient.swift [hunk 1]
```diff
@@ -1,31 +1,121 @@
...
+
+    public func respond(_ params: RespondParams) async throws {
+        guard let account = account else { throw Errors.unknownWalletAddress }
+        try await walletRespondService.respond(respondParams: params, account: account)
+    }
+
+    public func getPendingRequests() throws -> [AuthRequest] {
+        guard let account = account else { throw Errors.unknownWalletAddress }
+        let pendingRequests: [AuthRequest] = rpcHistory.getPending()
+            .filter {$0.request.method == "wc_authRequest"}
```
Comment (flypaper0) [inline Sources/Auth/AuthClient.swift:94]: mb move filter and mapping out of AuthClient? rpcHistoryService or pendingService

File: Sources/Auth/AuthClient.swift [hunk 2]
```diff
@@ -1,31 +1,121 @@
...
 
-    func pair(uri: String) async throws {
+    public func pair(uri: String) async throws {
         guard let pairingURI = WalletConnectURI(string: uri) else {
             throw Errors.malformedPairingURI
         }
         try await walletPairService.pair(pairingURI)
     }
+
+    public func request(_ params: RequestParams, topic: String?) async throws -> String? {
```
Comment (flypaper0) [inline Sources/Auth/AuthClient.swift:70]: let's split it into two methods?

File: Sources/Auth/AuthClient.swift (hunk 1 above)
Comment (llbartekll) [reply Sources/Auth/AuthClient.swift:94]: good idea!

File: Sources/Auth/AuthClient.swift (hunk 2 above)
Comment (llbartekll) [reply Sources/Auth/AuthClient.swift:70]: 🤔

**Review by flypaper0 — APPROVED** [review]

File: Sources/Auth/AuthClient.swift (hunk 2 above)
Comment (llbartekll) [reply Sources/Auth/AuthClient.swift:70]: actually I agree, looks better

**Pull request merged**